test/integration/test_integration.py

```diff
@@ -1479,5 +1479,62 @@ def forward(self, x):
         torch._export.aot_compile(model, example_inputs)
 
 
+class TestExport(unittest.TestCase):
+    @parameterized.expand(
+        list(itertools.product(TENSOR_SUBCLASS_APIS, COMMON_DEVICES, COMMON_DTYPES)),
+    )
+    def test_aoti(self, api, test_device, test_dtype):
+        if not TORCH_VERSION_AFTER_2_4:
+            self.skipTest("aoti compatibility requires 2.4+.")
+
+        # if test_device == "cuda":
+        #     self.skipTest("AOTI has some issues in cuda test right now, skipping")
+
+        logger.info(f"TestExport: {api}, {test_device}, {test_dtype}")
+        # if api is change_linear_weights_to_int8_dqtensors and test_device == "cuda":
+        #     self.skipTest(f"{api} in {test_device} is not support for aoti compilation yet")
+
+        if test_dtype != torch.bfloat16:
+            self.skipTest(f"{api} in {test_dtype} is not support for aoti compilation yet")
+
+        if test_device == "cuda" and not torch.cuda.is_available():
+            self.skipTest(f"Need CUDA available.")
+        if test_device == "cuda" and torch.cuda.is_available() and test_dtype == torch.bfloat16 and torch.cuda.get_device_capability() < (8, 0):
```

Comment:
more of a nit but can you move these to the top of the function with `unittest.skipif()` I'm not used to reading a function to figure out what gets skipped

Reply:
right, these are kind of deepened on the args, we do have a (brutal) helper function that does this, but I need to think about how we can make that more robust: https://github.com/pytorch/ao/blob/main/test/integration/test_integration.py#L98